tests/unit_tests/test_serdes.py

```diff
@@ -90,6 +94,7 @@ class DeAndSerializationTestCase:
     DeAndSerializationTestCase({"foo": 1}, b'{"foo":1}', dict),
     DeAndSerializationTestCase(Foo(bar=1, baz="baz"), b'{"bar":1,"baz":"baz"}', Foo),
     DeAndSerializationTestCase(np.array([1, 2, 3]), b"[1,2,3]", list),
+    DeAndSerializationTestCase(Bar(foo={1, 2, 3}), b'{"foo":[1,2,3]}', Bar),
```

Comment:
Can we also do the other test case from https://github.com/DiamondLightSource/bluesky-stomp/pull/35/files?

Reply:
Added